File: mobile/src/main/java/jp/satorufujiwara/kotlin/ui/main/MainModule.java
package jp.satorufujiwara.kotlin.ui.main;

import android.app.Application;

import java.util.List;

import dagger.Module;
import dagger.Provides;
import jp.satorufujiwara.kotlin.data.api.GitHubService;
import jp.satorufujiwara.kotlin.data.api.dto.Repo;
import rx.Observable;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

@Module
public class MainModule {

    @Provides
    Observable<List<? extends Repo>> provideGithubObservable(GitHubService gitHubService) {
        return gitHubService.listRepos("octcat")
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
    }

    @Provides
    Repo provideRepo(Application application) {
        return new Repo("id", "name");
    }

}

File: mobile/src/main/kotlin/jp/satorufujiwara/kotlin/ui/main/MainModule.kt
package jp.satorufujiwara.kotlin.ui.main

import com.squareup.sqlbrite.BriteDatabase
import dagger.Module
import dagger.Provides
import jp.satorufujiwara.kotlin.data.repository.GitHubRepository

@Module
class MainModule {

    @Provides
    @MainScope
    fun provideMainDispatcher(db: BriteDatabase) = MainDispatcher(db)

    @Provides
    @MainScope
    fun provideMainAction(dispatcher: MainDispatcher, repository: GitHubRepository) =
            MainAction(dispatcher, repository)

    @Provides
    @MainScope
    fun provideMainStore(dispatcher: MainDispatcher) = MainStore(dispatcher)
}
